packages/dev/core/src/Meshes/abstractMesh.hotSpot.ts

```diff
@@ -0,0 +1,119 @@
+import { Vector2, Vector3, TmpVectors, Matrix } from "../Maths/math.vector";
+import type { AbstractMesh } from "./abstractMesh";
+import { VertexBuffer } from "../Buffers/buffer";
+
+/**
+ * Data for mesh hotspot computation
+ */
+export class HotSpotQuery {
+    /**
+     * 3 point indices
+     */
+    pointIndex: [number, number, number];
+    /**
+     * 3 barycentric coordinates
+     */
+    barycentric: [number, number, number];
+}
+
+/**
+ * Result from a HotSpot query on the scene
+ */
+export class HotSpot {
+    /**
+     * 2D screen position
+     */
+    screenPosition = new Vector2();
+    /**
+     * 3D world position
+     */
+    worldPosition = new Vector3();
+}

```

Comment:
Same thing here, I think this should just be:
```suggestion
export type HotSpot {
    /**
     * 2D screen position
     */
    screenPosition: Vector2;
    /**
     * 3D world position
     */
    worldPosition: Vector3;
}

```

Reply:
It means having to instanciate the vector2 and vector3 in index.html for `hotspotResult` ?